File: Neuroflow/NeuralNetworks/_LearningAlgo/LearningBehavior.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Neuroflow.NeuralNetworks
{
    public abstract class LearningBehavior : LayerBehavior
    {
        public int GroupID { get; set; }

        protected override bool PropsEquals(LayerBehavior other)
        {
            return GroupID == ((LearningBehavior)other).GroupID;
        }

        protected override int GenerateHashCode()
        {
            return GroupID;
        }
    }
}

File: NeuroflowLib/learning_behavior.cpp
#include "stdafx.h"
#include "learning_behavior.h"

USING;

bool learning_behavior::props_equals(const layer_behavior_ptr& other) const
{
    auto o = dynamic_pointer_cast<learning_behavior>(other);
    assert(o);
    return group_id == o->group_id;
}